main.go

```diff
@@ -92,7 +98,6 @@ func init() {
 }
 
 var (
-	metricsAddr                        string
```

Comment:
These changes are mirrored from #4182 to account for breaking changes in controller-runtime 0.16, which ASO 2.4 has a hard dependency on.